src/megaclient.cpp

```diff
@@ -9326,7 +9326,7 @@ void MegaClient::notifynode(Node* n)
         // deletion from LocalNodes.
         if (n->localnode && n->localnode->parent && n->parent && !n->parent->localnode)
         {
-            if (n->changed.removed || n->changed.parent)
+            if (n->changed.removed || n->changed.newnode)
```

Comment:
I think you actually want to keep the code as it used to be here. The callbacks below are for nodes that are moved outside of the sync --> parent has changed.
```suggestion
            if (n->changed.removed || n->changed.parent)
```